cwms-data-api/src/main/java/cwms/cda/data/dao/texttimeseries/StandardTimeSeriesTextDao.java

```diff
@@ -0,0 +1,312 @@
+
+        try (ResultSet rs = CWMS_TEXT_PACKAGE.call_CAT_STD_TEXT_F(dsl.configuration(),
+                stdTextIdMask, officeIdMask).intoResultSet()) {
+
+            return buildCatalog(rs);
+        }
+
+    }
+
+    private static StandardTextCatalog buildCatalog(ResultSet rs) throws SQLException {
+        OracleTypeMap.checkMetaData(rs.getMetaData(), stdTextCatalogColumnsList, CATALOG_TYPE);
+        StandardTextCatalog.Builder builder = new StandardTextCatalog.Builder();
+        while (rs.next()) {
+            builder.withValue(buildStandardTextValue(
+                    rs.getString(OFFICE_ID), rs.getString(STD_TEXT_ID), rs.getString(STD_TEXT)));
+        }
+        return builder.build();
+    }
+
+    private static StandardTextValue buildStandardTextValue(String officeId, String txtId, String txt) {
+        StandardTextId id = new StandardTextId.Builder()
+                        .withOfficeId(officeId)
+                        .withId(txtId)
+                        .build();
+        return new StandardTextValue.Builder()
+                .withId(id)
+                .withStandardText(txt)
+                .build();
+    }
+
+
+    public StandardTextValue retrieve(StandardTextId standardTextId) {
+        return connectionResult(dsl, c -> retrieve(c, standardTextId));
+    }
+
+    private StandardTextValue retrieve(Connection c, StandardTextId standardTextId) throws SQLException {
+        CwmsDbText dbText = CwmsDbServiceLookup.buildCwmsDb(CwmsDbText.class, c);
+
+        String stdTextClob = dbText.retrieveStdTextF(c,
+                standardTextId.getId(),
+                standardTextId.getOfficeId());
+
+        return new StandardTextValue.Builder()
+                .withId(standardTextId)
+                .withStandardText(stdTextClob)
+                .build();
+    }
+
+    /**
+     * This is if you want to store a new standard text id -> value mapping.
+     * @param standardTextValue The standard text value to store
+     * @param failIfExists true if the store should fail if the standard text id already exists
+     */
+    public void store(StandardTextValue standardTextValue, boolean failIfExists) {
+        StandardTextId standardTextId = standardTextValue.getId();
+        String stdTextId = standardTextId.getId();
+        String stdText = standardTextValue.getStandardText();
+        String officeId = standardTextId.getOfficeId();
+
+        connection(dsl, c -> {
+            setOffice(c, officeId);
+            CwmsDbText dbText = CwmsDbServiceLookup.buildCwmsDb(CwmsDbText.class, c);
+            dbText.storeStdText(c, stdTextId, stdText, failIfExists, officeId);
+        });
+    }
+
+    public void storeRows(String officeId, String tsId, boolean maxVersion, boolean replaceAll, Collection<StandardTextTimeSeriesRow> stdRows) {
+        for (StandardTextTimeSeriesRow stdRow : stdRows) {
+            store(officeId, tsId, stdRow, maxVersion, replaceAll);
```

Comment:
Parameterize in the Connection so that we aren't pulling a new connection from the pull on every row